client/state/sites/guided-transfer/actions.js

```diff
@@ -61,3 +64,42 @@ export function requestGuidedTransferStatus( siteId ) {
 			.catch( failure );
 	};
 }
+
+/**
+ * Saves a user's target host details in preparation for
+ * a guided transfer to that host
+ *
+ * @param {number} siteId The id of the source site to transfer
+ * @param {Object} data The form data containing the target host details
+ * @returns {Thunk} Action thunk
+ */
+export function saveHostDetails( siteId, data ) {
+	return ( dispatch ) => {
+		dispatch( {
+			type: GUIDED_TRANSFER_HOST_DETAILS_SAVE,
+			siteId,
+		} );
+
+		const success = response => {
+			dispatch( {
+				type: GUIDED_TRANSFER_HOST_DETAILS_SAVE_SUCCESS,
+				siteId,
+			} );
+
+			// The success response is the updated status of the guided transfer
+			dispatch( receiveGuidedTransferStatus(
+				siteId, omit( response, '_headers' ) )
+			);
```

Comment:
I think that trailing `)` on the parameters should be on this line instead so that it looks like
```js
dispatch( receiveGuidedTransferStatus(
 	siteId, omit( response, '_headers' )
) );
```